rn/Teacher/src/modules/discussions/details/EditReply.js

```diff
@@ -141,10 +120,22 @@ export class EditReply extends React.Component<Props, any> {
       attachment: this.state.attachment,
     }
     this.setState({ pending: true })
-    if (this.props.isEdit) {
-      this.props.editEntry(this.props.context, this.props.contextID, this.props.discussionID, this.props.entryID, params, this.props.indexPath)
-    } else {
-      this.props.createEntry(this.props.context, this.props.contextID, this.props.discussionID, this.props.entryID, params, this.props.indexPath, this.props.lastReplyAt)
+    const isNew = !this.props.isEdit
+    try {
+      if (isNew) {
+        await this.props.createEntry(this.props.context, this.props.contextID, this.props.discussionID, this.props.entryID, params, this.props.indexPath, this.props.lastReplyAt).promise
```

Comment:
The modal is still dismissing even when there is an error. I checked it out locally and I was getting `undefined` for this action's `.promise`.

I think it's because we are deleting the promise in our middleware [here](https://github.com/instructure/canvas-ios/blob/master/rn/Teacher/src/redux/middleware/redux-promise.js#L65).